apps/charterafrica/src/components/Datasets/DatasetCard.js

```diff
@@ -41,11 +39,7 @@ function DatasetCard({
         ...sx,
       }}
     >
-      <CardActionArea
-        component={Link}
-        // TODO: Update URL to single dataset page
-        href={url}
-      >
+      <CardActionArea component={Link} href={`${pageUrl}/${id}`}>
```

Comment:
Just create a `href` prop when formatting the dataset.

Reply:
@kilemensi This will require us to use a hardcoded string `/resources/datasets/` when formatting datasets. Won't this cause an issue if for example the `dataset `block is used in a different page?

Reply:
@kilemensi Not sure I get you. Here is my quick view:
Say we are on the page `/resources/datasets` if we click on one of the datasets, it should take us to `/resources/datasets/id`. We can easily add this in the `processPageDatasets` function, but this would not work as subsequent calls to `/api/v1/resources/datasets` would not know which page it was called from to append the id of each dataset.
To add the individual URL in each dataset in the [formatDataset](https://github.com/CodeForAfrica/ui/blob/fb3692fe328c3210e46d2d210ebefb1f958a19d6/apps/charterafrica/src/lib/openAfrica.js#L46) function, we'll need to pass around the url to append the dataset id to, both in the `processPageDatasets` funtion and the `useDatasets` hook, which I think is quite unnecessary as it is not information required to fetch data from openAfrica.

Reply:
3. Every `req` on API side also comes with payload i.e. `req.payload`. You can load the `datasets` page as well without requiring anything passed-in from the front-end.